protocol/contracts/amo/helpers/PoolHelper.sol

```diff
@@ -335,6 +342,76 @@ contract PoolHelper {
         }
     }
 
+    // @notice Get the quote used to add liquidity proportionally
+    // @dev Since this is not the view function, this should be called with `callStatic`
+    function proportionalAddLiquidityQuote(
+        uint256 stablesAmount,
+        uint256 stableWeightBps,
+        uint256 slippageBps
+    ) external returns (
+        uint256 templeAmount,
+        uint256 expectedBptAmount,
+        uint256 minBptAmount,
+        AMO__IBalancerVault.JoinPoolRequest memory requestData
+    ) {
+        (uint256 templeBalanceInLP, uint256 stableBalanceInLP) = getTempleStableBalances();
+        templeAmount = stableBalanceInLP == 0
+            ? stablesAmount / stableWeightBps * (TPF_PRECISION - stableWeightBps)
```

Comment:
Hmm... I think the weight will not affect the adding/removing liquidity and we don't need to use the `stableWeightBps`. I think we can calculate the proportional `templeAmount` using `(templeBalanceInLP * stablesAmount / stableBalanceInLP)`.
So maybe ```
templeAmount = stableBalanceInLP == 0
            ? stablesAmount
            : (templeBalanceInLP * stablesAmount / stableBalanceInLP);
```

Reply:
I *think* this is the logic which does it
https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/pool-weighted/contracts/WeightedMath.sol#L78

Called from:
https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/pool-weighted/contracts/BaseWeightedPool.sol#L111

Reply:
Ah actually - the balancer sdk is a good resource to check too:
https://github.com/balancer/balancer-sdk/blob/be692be5d6057f5e44362667e47bd7ecf9a83b37/balancer-js/src/modules/pools/proportional-amounts/index.ts#L24

Reply:
And just for some more context because i was poking around - the swap price calc for weighted math in typescript:
https://github.com/balancer/balancer-sor/blob/7b087e32cb9d3d24af301829b169ac88e67f1b54/src/pools/weightedPool/weightedMath.ts#L483